File: 977_Squares_of_Sorted_Arrays/Solution.py
class Solution(object):
    def sortedSquares(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        # this is two pinter approach
        result = []
        l = 0
        r = len(nums)-1
        while l <= r:
            if nums[l]**2 < nums[r]**2:
                result.append(nums[r]**2)
                r -= 1
            else:
                result.append(nums[l]**2)
                l += 1
        return result[::-1]  # returns result in negative order

File: 977_Squares_of_Sorted_Arrays/Solution.java
import java.util.*;

class Solution {
    public int[] sortedSquares(int[] nums) {
        int length = nums.length;
        int[] result = new int[length];
        int begin = 0;
        int end = length-1;
        int i = length-1;
        while (i>=0){
            if (nums[begin]*nums[begin] > nums[end]*nums[end]) {
                result[i] = (nums[begin]*nums[begin]);
                begin+=1;
                i--;
            }
            else{
                result[i] = (nums[end]*nums[end]);
            
                end-=1;
                i--;

            }
                
        }
        return result;

    }
}